app/screens/convert_gm_to_channel/convert_gm_to_channel_form/convert_gm_to_channel_form.tsx

```diff
@@ -0,0 +1,137 @@
+            paddingHorizontal: 20,
+            display: 'flex',
+            flexDirection: 'column',
+            gap: 24,
+        },
+    };
+});
+
+type Props = {
+    channelId: string;
+    commonTeams: Team[];
+    profiles: UserProfile[];
+    locale?: string;
+    teammateNameDisplay?: string;
+}
+
+export const ConvertGMToChannelForm = ({
+    channelId,
+    commonTeams,
+    profiles,
+    locale,
+    teammateNameDisplay,
+}: Props) => {
+    const theme = useTheme();
+    const styles = getStyleFromTheme(theme);
+    const serverUrl = useServerUrl();
+    const intl = useIntl();
+
+    const [selectedTeam, setSelectedTeam] = useState<Team>();
+    const [newChannelName, setNewChannelName] = useState<string>('');
+
+    const submitButtonEnabled = selectedTeam && newChannelName.trim();
+
+    const {formatMessage} = useIntl();
+    const confirmButtonText = formatMessage({
+        id: 'channel_info.convert_gm_to_channel.button_text',
+        defaultMessage: 'Convert to Private Channel',
+    });
+
+    useEffect(() => {
+        if (commonTeams.length > 0) {
+            setSelectedTeam(commonTeams[0]);
+        }
+    }, [commonTeams]);
+
+    const handleOnPress = useCallback(async () => {
+        if (!submitButtonEnabled) {
+            return;
+        }
+
+        const {updatedChannel, error} = await convertGroupMessageToPrivateChannel(serverUrl, channelId, selectedTeam.id, newChannelName);
+
+        if (error) {
+            logError(error);
+            return;
+        }
+
+        if (!updatedChannel) {
+            return;
+        }
+
+        await switchToChannelById(serverUrl, updatedChannel.id, selectedTeam.id);
+    }, [selectedTeam, newChannelName, submitButtonEnabled]);
+
+    const messageBoxHeader = intl.formatMessage({
+        id: 'channel_info.convert_gm_to_channel.warning.header',
+        defaultMessage: 'Conversation history will be visible to any channel members',
+    });
+
+    const userDisplayNames = profiles.map((profile) => displayUsername(profile, locale, teammateNameDisplay));
+    const defaultUserDisplayNames = intl.formatMessage({id: 'channel_info.convert_gm_to_channel.warning.body.yourself', defaultMessage: 'yourself'});
+    const memberNames = profiles.length > 0 ? intl.formatList(userDisplayNames) : defaultUserDisplayNames;
+    const messageBoxBody = intl.formatMessage({
+        id: 'channel_info.convert_gm_to_channel.warning.bodyXXXX',
+        defaultMessage: 'You are about to convert the Group Message with {memberNames} to a Channel. This cannot be undone.',
+    }, {
+        memberNames,
+    });
+
+    if (commonTeams.length === 0) {
+        return (
+            <NoCommonTeamForm containerStyles={styles.container}/>
+        );
+    }
+
+    return (
+        <View style={styles.container}>
+            <MessageBox
+                header={messageBoxHeader}
+                body={messageBoxBody}
+            />
+            {
+                commonTeams.length > 1 &&
```

Comment:
if no multiple teams, does it make sense to display the name of the team that they share in common anyway but not allow them to change it ?

Reply:
I think it does, for both webapp and mobile. Checking with UX.

Reply:
Created https://mattermost.atlassian.net/browse/MM-55462 to handle this once we have something from UX. resolving for now.